File: server/models/QuizResult.py
from pydantic import BaseModel
from datetime import datetime
from typing import Dict, List
from models.UserAnswer import UserAnswer

class QuizResult(BaseModel):
    id: int
    score: int
    totalQuestions: int
    percentage: float
    createdAt: datetime
    userId: int
    answers: List[UserAnswer]
    user: Dict

class QuizResultAnswer(BaseModel):
    questionId: int
    answer: str
    isCorrect: bool

class QuizResultDto(BaseModel):
    score: int
    totalQuestions: int
    userId: int
    answers: List[QuizResultAnswer]

class QuizResultHistory(BaseModel):
    id: int
    score: int
    totalQuestions: int
    percentage: float
    createdAt: datetime

File: client/src/types/QuizResult.ts
export interface QuizResultAnswer {
    questionId: number
    answer: string
    isCorrect: boolean
}

export interface QuizResult {
    id: number
    score: number
    totalQuestions: number
    percentage: number
    userId: number
    answers: QuizResultAnswer[]
    createdAt: string
}